code/game/objects/items/rcd/RPD.dm

```diff
@@ -528,50 +528,57 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 			target_unsecured_pipe.update()
 
 	if(mode & BUILD_MODE)
+		var/beam
 		switch(category) //if we've gotten this var, the target is valid
 			if(ATMOS_CATEGORY) //Making pipes
 				if(!can_make_pipe)
 					return
 				playsound(get_turf(src), 'sound/machines/click.ogg', 50, TRUE)
+				if(bluespace)
+					beam = user.Beam(attack_target, icon_state = "rped_upgrade", time = atmos_build_speed)
 				if (recipe.type == /datum/pipe_info/meter)
-					if(do_after(user, atmos_build_speed, target = attack_target))
-						playsound(get_turf(src), RPD_USE_SOUND, 50, TRUE)
-						var/obj/item/pipe_meter/new_meter = new /obj/item/pipe_meter(get_turf(attack_target))
-						new_meter.setAttachLayer(piping_layer)
-						if(mode & WRENCH_MODE)
-							new_meter.wrench_act(user, src)
+					if(!do_after(user, atmos_build_speed, target = attack_target))
+						if(!isnull(beam))
+							qdel(beam)
```

Comment:
add return statement next line since `do_after()` has failed(outside the if condition)